test/mysql.test.js

```diff
@@ -235,6 +245,31 @@ describe('mysql', function() {
         });
       });
     });
+
+    it('isNewInstance should be undefined for after save hook', function(done) {
+      var student = {name: 'Joe', age: 20};
+      var newStudent = {};
+      Student.create(student, function(err, createdStudent) {
+        if (err) return done(err);
+        newStudent.id = createdStudent.id;
+        newStudent.name = 'Hannah';
+        newStudent.age = 25;
+        Student.replaceOrCreate(newStudent, function(err, s) {
+          if (err) return done(err);
+          s.id.should.equal(newStudent.id);
+          s.name.should.equal(newStudent.name);
+          s.age.should.equal(newStudent.age);
+        });
+
+        Student.observe('before save', function(ctx, next) {
```

Comment:
This is relying on the fact that before/after save hooks are invoked asynchronously, not in the same tick as `Student.replace()` was called. Also if these hook as never invoked, then the test will still pass.

A better solution is to:
 - register the hooks before calling `replaceOrCreate`
 - in the hooks, save the observed context objects
 - in `replaceOrCreate` callback, assert that context objects were stored and have expected `isNewInstance` value.